pkg/build/api/types.go

```diff
@@ -83,6 +83,11 @@ const (
 type BuildSource struct {
 	Type BuildSourceType `json:"type,omitempty"`
 	Git  *GitBuildSource `json:"git,omitempty"`
+
+	// Specify the sub-directory where the source code for the application exists.
+	// This allows to have buildable sources in directory other than root of
+	// repository.
+	ContextDir string `json:"contextDir,omitempty"`
```

Comment:
@bparees in that case they should be all removed ;-) I will make the current build finish and remove the json tagging.